airflow/cli/commands/dag_command.py

```diff
@@ -41,7 +41,7 @@ def _tabulate_dag_runs(dag_runs: List[DagRun], tablefmt="fancy_grid"):
             'Run ID': dag_run.run_id,
             'State': dag_run.state,
             'DAG ID': dag_run.dag_id,
-            'Execution date': dag_run.execution_date.isoformat(),
+            'Execution date': dag_run.execution_date.isoformat() if dag_run.execution_date else "",
```

Comment:
```suggestion
            'Execution date': dag_run.execution_date.isoformat() if dag_run.execution_date else None,
```
JSON supports null type, so we can use it instead of empty text.